### PR TITLE
schema refactors 1

### PR DESCRIPTION
**Pull request by AndrewCLu**

### CODE REVIEW

File: apps/jubmoji-quest/prisma/schema.prisma [hunk 1]
```diff
@@ -17,23 +17,30 @@ model Card {
...
+  id              Int      @id @default(autoincrement())
+  name            String
+  description     String
+  startTime       DateTime?
+  endTime         DateTime?
+  proofType       ProofType
+  proofParams     Json    // params for the specific proof type
+  imageLink       String?
+  isOfficial      Boolean @default(false)
+  isAlwaysVisible Boolean @default(false)
```
Comment (vb7401) [inline apps/jubmoji-quest/prisma/schema.prisma:42]: Maybe rename to `isHiddenUntilCollect` ?

File: apps/jubmoji-quest/prisma/schema.prisma [hunk 2]
```diff
@@ -51,18 +58,24 @@ model Power {
...
-  questId       Int  
-  quest         Quest   @relation(fields: [questId], references: [id])
-  qrCodes       QRCode[]   
-  sigNullifiers PowerSigNullifier[]
+  questId                   Int  
+  quest                     Quest   @relation(fields: [questId], references: [id])
+  prerequisiteCards         Card[]  @relation("PowerPrerequisiteCards") // cards needed to see the power
+  collectionCards           Card[]  @relation("PowerCollectionCards")   // cards needed to complete the power
+  qrCodes                   QRCode[]   
+  sigNullifiers             PowerSigNullifier[]
```
Comment (vb7401) [inline apps/jubmoji-quest/prisma/schema.prisma:71]: Need event details for QRCODE power type

File: apps/jubmoji-quest/prisma/schema.prisma [hunk 3]
```diff
@@ -51,18 +58,24 @@ model Power {
...
+  createdAt              DateTime @default(now())
 
   // Relationships
-  questId       Int  
-  quest         Quest   @relation(fields: [questId], references: [id])
-  qrCodes       QRCode[]   
-  sigNullifiers PowerSigNullifier[]
+  questId                   Int  
+  quest                     Quest   @relation(fields: [questId], references: [id])
+  prerequisiteCards         Card[]  @relation("PowerPrerequisiteCards") // cards needed to see the power
```
Comment (vb7401) [inline apps/jubmoji-quest/prisma/schema.prisma:68]: Shouldn't these two relations be optional? for the exitsing powers that don't have these

**Review by vb7401 — APPROVED** [review]
LGTM after these comments

File: apps/jubmoji-quest/prisma/schema.prisma (hunk 3 above)
Comment (AndrewCLu) [reply apps/jubmoji-quest/prisma/schema.prisma:68]: there is no Card[]? type

**Pull request merged**